### PR TITLE
SAK-45517 - signup jsf upgrade

### PR DESCRIPTION
**Pull request by maurercw**

### CODE REVIEW

File: signup/tool/src/webapp/js/signupScript.js [hunk 1]
```diff
@@ -124,9 +124,14 @@
...
-			var groupId = groupIdPrefix+i+groupIdSuffix;
-			document.getElementById(groupId).disabled=!enable;
+
+		// jsf now renders a row with empty td elements even when there is no data
+		var cellData = document.getElementById("meeting:currentSiteGroups").tBodies[0].rows[0].cells[0].innerHTML;
+		if (cellData != "") {
+            for(var i=0; i<size; i++){
+                var groupId = groupIdPrefix+i+groupIdSuffix;
+                document.getElementById(groupId).disabled=!enable;
+            }
```
Comment (bjones86) [inline signup/tool/src/webapp/js/signupScript.js:134]: spacing is a little off here

File: signup/tool/src/webapp/js/signupScript.js [hunk 2]
```diff
@@ -136,9 +141,14 @@
...
-			var siteId = sitePrefix + i + siteSuffix;
-			enableOtherGroupsSelection(sitePrefix + i, !document.getElementById(siteId).checked);
+
+		// jsf now renders a row with empty td elements even when there is no data
+        var cellData = document.getElementById(siteId).tBodies[0].rows[0].cells[0].innerHTML;
+        if (cellData != "") {
+            for(var i=0; i<numberOfSites; i++){
+                var siteId = sitePrefix + i + siteSuffix;
+                enableOtherGroupsSelection(sitePrefix + i, !document.getElementById(siteId).checked);
+            }
```
Comment (bjones86) [inline signup/tool/src/webapp/js/signupScript.js:151]: and here

File: signup/tool/src/webapp/js/signupScript.js [hunk 3]
```diff
@@ -147,9 +157,14 @@
...
-			var groupId = groupIdPrefix+i+groupIdSuffix;
-			document.getElementById(groupId).disabled=!enable;
+
+		// jsf now renders a row with empty td elements even when there is no data
+        var cellData = document.getElementById(prefixId+":userGroups").tBodies[0].rows[0].cells[0].innerHTML;
+        if (cellData != "") {
+            for(var i=0; i<size; i++){
+                var groupId = groupIdPrefix+i+groupIdSuffix;
+                document.getElementById(groupId).disabled=!enable;
+            }
```
Comment (bjones86) [inline signup/tool/src/webapp/js/signupScript.js:167]: and here

**Review by bjones86 — CHANGES_REQUESTED** [review]
LGTM, just some minor issues

File: signup/tool/src/webapp/js/signupScript.js (hunk 1 above)
Comment (maurercw) [reply signup/tool/src/webapp/js/signupScript.js]: should be fixed

File: signup/tool/src/webapp/js/signupScript.js (hunk 2 above)
Comment (maurercw) [reply signup/tool/src/webapp/js/signupScript.js]: should be fixed

File: signup/tool/src/webapp/js/signupScript.js (hunk 3 above)
Comment (maurercw) [reply signup/tool/src/webapp/js/signupScript.js]: should be fixed

**Review by ern — APPROVED** [review]

**Review by bjones86 — APPROVED** [review]

**Pull request merged**